create_salsa_project.go

```diff
@@ -0,0 +1,166 @@
+package main
+
+import (
+	"encoding/json"
+	"errors"
+	"flag"
+	"fmt"
+	"io/ioutil"
+	"log"
+	"net/http"
+	"net/url"
+	"os"
+	"strconv"
+)
+
+const (
+	salsaGroupID = 2638
+	salsaApiUrl  = "https://salsa.debian.org/api/v4"
+)
+
+type gitlabProject struct {
+	ID            int    `json:"id"`
+	Name          string `json:"name"`
+	HttpUrlToRepo string `json:"http_url_to_repo"`
+}
+
+func execCreateSalsaProject(args []string) {
+	projectName := getProjectNameOrExit(args)
+	token := getSalsaTokenOrExit()
+
+	gitlabProject, err := createSalsaProject(projectName, token)
+
+	if err != nil {
+		fmt.Fprintf(os.Stderr, "Could not create %s on salsa: %s\n", projectName, err)
+		os.Exit(1)
+	}
+
+	fmt.Printf(
+		"Project %s was created on salsa.debian.org: %s\n",
+		gitlabProject.Name,
+		gitlabProject.HttpUrlToRepo,
+	)
+
+	err = createSalsaWebhook(
+		gitlabProject.ID,
+		"https://webhook.salsa.debian.org/tagpending/"+gitlabProject.Name,
+		token,
+	)
+
+	if err != nil {
+		fmt.Fprintf(os.Stderr, "Could not create webhook on salsa project: %s\n", err)
+		os.Exit(1)
+	}
+
+}
+
+func postFormToSalsaApi(path string, data url.Values, token string) (*http.Response, error) {
+	postUrl := salsaApiUrl + path
+	data.Add("private_token", token)
+	return http.PostForm(postUrl, data)
+}
+
+func createSalsaWebhook(projectId int, webhookUrl, token string) error {
+	postPath := "/projects/" + strconv.Itoa(projectId) + "/hooks"
+
+	response, err := postFormToSalsaApi(
+		postPath,
+		url.Values{
+			"url":         {webhookUrl},
+			"push_events": {"true"},
+		},
+		token,
+	)
+
+	if err != nil {
+		return err
+	}
+
+	if response.StatusCode != http.StatusCreated {
+		responseData, err := ioutil.ReadAll(response.Body)
+		if err != nil {
+			return err
+		}
+
+		return errors.New(
```

Comment:
Use fmt.Errorf instead of errors.New+fmt.Sprintf

Reply:
Done!